### PR TITLE
Add option to invalidate duties based on the dependent root instead of chain reorg events

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
Adds a hidden option to switch from using chain re-org events to the dependent root information to detect when duties need to be invalidated.

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/AbstractDutyScheduler.java [hunk 1]
```diff
@@ -54,8 +59,41 @@ public void onSlot(final UInt64 slot) {
...
+    if (!useDependentRoots) {
+      return;
+    }
+    final UInt64 headEpoch = compute_epoch_at_slot(slot);
+    dutiesByEpoch
+        .tailMap(headEpoch, true)
+        .forEach(
+            (dutyEpoch, duties) ->
+                duties.onHeadUpdate(
+                    getExpectedTargetRoot(
```
Comment (mbaxter) [inline validator/client/src/main/java/tech/pegasys/teku/validator/client/AbstractDutyScheduler.java]: Should this be:
```suggestion
                    getExpectedDependentRoot(
```
?

Seems like "dependent" and "target" roots are equivalent, if so would clarify things to use consistent naming.

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/BlockDutyScheduler.java [hunk 2]
```diff
@@ -46,4 +49,14 @@ public void onBlockProductionDue(final UInt64 slot) {
...
   }
+
+  @Override
+  protected Bytes32 getExpectedTargetRoot(
+      final Bytes32 headBlockRoot,
+      final Bytes32 previousTargetRoot,
+      final Bytes32 currentTargetRoot,
+      final UInt64 headEpoch,
+      final UInt64 dutyEpoch) {
+    return headEpoch.equals(dutyEpoch) ? currentTargetRoot : headBlockRoot;
```
Comment (mbaxter) [inline validator/client/src/main/java/tech/pegasys/teku/validator/client/BlockDutyScheduler.java:67]: Looks like these methods assume that dutyEpoch >= headEpoch, maybe we should assert that here to be safer / clearer.

**Review by mbaxter — APPROVED** [review]
LGTM

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/AbstractDutyScheduler.java (hunk 1 above)
Comment (ajsutton) [reply validator/client/src/main/java/tech/pegasys/teku/validator/client/AbstractDutyScheduler.java]: Yep, the original proposal from Paul H called them target roots and they got renamed to dependent roots as part of refining how they fit into the REST API, but that was after I did the original draft PR - apparently I missed renaming this when bringing it over.

File: validator/client/src/main/java/tech/pegasys/teku/validator/client/BlockDutyScheduler.java (hunk 2 above)
Comment (ajsutton) [reply validator/client/src/main/java/tech/pegasys/teku/validator/client/BlockDutyScheduler.java:67]: Yep, it's called after doing a tailMap so always guaranteed to meet that condition - checkArgument added.

**Pull request merged**